v2/options.go

```diff
@@ -2,51 +2,57 @@ package car
 
 import "github.com/multiformats/go-multicodec"
 
-// ReadOptions holds the configured options after applying a number of
-// ReadOption funcs.
-//
-// This type should not be used directly by end users; it's only exposed as a
-// side effect of ReadOption.
-type ReadOptions struct {
-	ZeroLengthSectionAsEOF bool
+// Option describes an option which affects behavior when interacting with CAR files.
+type Option func(*Options)
 
-	BlockstoreUseWholeCIDs bool
-}
+func (Option) readWriteOption() {}
 
-// ApplyReadOptions applies ropts and returns the resulting ReadOptions.
-func ApplyReadOptions(ropts ...ReadOption) ReadOptions {
-	var opts ReadOptions
-	for _, opt := range ropts {
-		opt(&opts)
-	}
-	return opts
-}
+var _ ReadWriteOption = Option(nil)
 
-// ReadOption describes an option which affects behavior when parsing CAR files.
-type ReadOption func(*ReadOptions)
+// ReadOption hints that an API wants options related only to reading CAR files.
+// Deprecated: use Option instead.
+type ReadOption Option
 
 func (ReadOption) readWriteOption() {}
 
 var _ ReadWriteOption = ReadOption(nil)
 
-// WriteOptions holds the configured options after applying a number of
-// WriteOption funcs.
+// WriteOption hints that an API wants options related only to reading CAR files.
+// Deprecated: use Option instead.
```

Comment:
I'm not sure I would deprecate ReadOption and WriteOption. If an API *only* reads or *only* writes, taking one of those two will then help make the API clearer, even if it's just for the sake of names and docs.